services/app-api/src/postgres/contractAndRates/parseContractWithHistory.ts

```diff
@@ -157,15 +157,33 @@ function contractWithHistoryToDomainModel(
                 )
             }
 
-            draftRevision = draftContractRevToDomainModel(contractRev)
+            const domainRateRevisions: RateRevisionType[] = []
+            for (const submission of contractRev.relatedSubmisions) {
+                const relatedRateRevisions = submission.submissionPackages
+                    .filter((p) => p.contractRevisionID === contractRev.id)
+                    .sort((a, b) => a.ratePosition - b.ratePosition)
+                    .map((p) => p.rateRevision)
 
-            if (draftRevision instanceof Error) {
-                return new Error(
-                    `error converting draft contract revision with id ${contractRev.id} to domain model: ${draftRevision}`
-                )
+                for (const rateRev of relatedRateRevisions) {
+                    const domainRate = rateRevisionToDomainModel(rateRev)
+
+                    domainRateRevisions.push(domainRate)
+                }
+            }
+
+            draftRevision = {
+                id: contractRev.id,
+                contract: contractRev.contract,
+                createdAt: contractRev.createdAt,
+                updatedAt: contractRev.updatedAt,
+                unlockInfo: convertUpdateInfoToDomainModel(
+                    contractRev.unlockInfo
+                ),
+                formData: contractFormDataToDomainModel(contractRev),
+                rateRevisions: domainRateRevisions,
```

Comment:
I think we should drop rateRevisions from the domain model, we should just use contract.draftRates instead.

Reply:
Okay, I can work on those domain model changes. @macrael I'm assuming make the change symetrically both contracts and rates have this pattern in their domain model

Reply:
Update: Can't make this change symetrically because it causes a types infinity loop. So made the change for contracts (drop rate revisions from the draftRevision domain model)  only.